overlord/hookstate/hookmgr.go

```diff
@@ -253,24 +254,38 @@ func (m *HookManager) GracefullyWaitRunningHooks() bool {
 // Note that this method is synchronous, as the task is already running in a
 // goroutine.
 func (m *HookManager) doRunHook(task *state.Task, tomb *tomb.Tomb) error {
-	task.State().Lock()
+	st := task.State()
+	st.Lock()
+	defer task.State().Unlock()
+
 	hooksup, snapst, err := hookSetup(task, "hook-setup")
-	task.State().Unlock()
 	if err != nil {
 		return fmt.Errorf("cannot extract hook setup from task: %s", err)
 	}
 
-	return m.runHook(task, tomb, snapst, hooksup)
+	perfTimings := timings.NewForTask(task)
+	// XXX: overwrite task-kind with hook name, e.g. "run-hook[install]"; should we use separate tag?
+	perfTimings.AddTag("task-kind", fmt.Sprintf("%s[%s]", task.Kind(), hooksup.Hook))
+
+	st.Unlock()
+	timings.Run(perfTimings, "execute-hook", fmt.Sprintf("execute hook %q of snap %q", hooksup.Hook, hooksup.Snap), func(timings.Measurer) {
```

Comment:
there's not a lot going on here outside the invoked m.RunHook function,  a bit unclear why this would be much different than the timing for the overall task? in other words do we need this?

Reply:
Fair point. The main motivation was to provide more context (hook name) that we don't have when simply reporting run-hook tasks. I'll remove this for now and we can discuss this later as a potential followup, perhaps we could just special-case the output for "run-hook" tasks.